### ISSUE TITLE
Indices identifier are not slugified

### ISSUE THREAD

**Issue by huard** [Ouranosinc/xclim#170]
* xclim version: master

### Description

The identifier attribute of indices should be unique and I think should also be a valid variable name (no . - characters) so it can be used as an HTML anchor, a python variable name, etc. Currently all identifiers using thresholds with negative numbers or floats would cause problems. 

Travis, has this been an issue so far for you ?

**Comment by tlogan2000** [comment]
This hasn't been a deal breaker .... OpenDap links seem to work ok when hypens are present
However, it does restrict methods for accessing the data variables in a xarray dataset (see below)

This could also potentially cause issues with other packages reading .nc files? netCDF4 others ?...

```
import xarray as xr

url = 'http://boreas.ouranos.ca:8083/thredds/dodsC/birdhouse/cccs_portal/indices/Final/BCCAQv2/tnlt_-25/YS/rcp85/simulations/BCCAQv2+ANUSPLIN300_inmcm4_historical+rcp85_r1i1p1_1950-2100_tnlt_-25_YS.nc'
ds = xr.open_dataset(url, decode_times=False)
print(ds)
print(ds['tnlt_-25']) OK
print(ds.tnlt_-25) # does not work

**Comment by tlogan2000** [comment]
using netcdf4 directly seems ok as well...  

```
import netCDF4 as nc
import numpy as np
url = 'http://boreas.ouranos.ca:8083/thredds/dodsC/birdhouse/cccs_portal/indices/Final/BCCAQv2/tnlt_-25/YS/rcp85/simulations/BCCAQv2+ANUSPLIN300_inmcm4_historical+rcp85_r1i1p1_1950-2100_tnlt_-25_YS.nc'

ds = nc.Dataset(url)
print(np.max(ds['tnlt_-25'][0,:,:]))

**Comment by huard** [comment]
Kinda fixed in #216
